### PR TITLE
feat(data): pricing source of truth, fixture invariants, screenshot timestamps

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary

Creates `constants/plans.ts` as the canonical pricing source of truth for Jovie. Removes all "waitlist" and "Request Access" copy from marketing pricing pages. Adds contract tests, fixture invariants, and screenshot timestamp invariants.

**Linear issues:** JOV-2079, JOV-2077, JOV-2078, JOV-2087
**Version:** 26.4.236

### New files (HOT ZONE)
- `apps/web/constants/plans.ts` — `CANONICAL_PLANS` export: display names, prices, CTA labels/hrefs for free/pro/max. Single source of truth for marketing + onboarding.
- `apps/web/tests/unit/pricing/pricing-source-of-truth.test.ts` — 16 contract tests: plan IDs, prices, banned phrases, signup href `?plan=` params
- `apps/web/tests/unit/seed/seed-fixture-invariants.test.ts` — 9 fixture invariant tests: no placeholder titles, correct artist credits, Spotify ID + artwork pairing
- `apps/web/tests/unit/product-screenshots/screenshot-player-timestamps.test.ts` — 4 timestamp invariant tests: unique player timestamps, valid format

### Modified files (HOT ZONE)
- `apps/web/app/(marketing)/pricing/page.tsx` — removed waitlist paragraph, updated CTAs, InStock availability in JSON-LD
- `apps/web/app/(marketing)/launch/pricing/page.tsx` — CTAs now use `getCanonicalPlan()` for labels/hrefs (no drift)
- `apps/web/components/features/pricing/MarketingPricingPlans.tsx` — removed waitlist helpers, CTA from data layer
- `apps/web/data/marketingPricingPlans.ts` — replaced team/enterprise with max; added `ctaLabel`/`ctaHref`; prices from `PLAN_PRICES`
- `apps/web/lib/screenshots/types.ts` — added `playerTimestamp?: string` to `ScreenshotScenario`
- `apps/web/lib/screenshots/registry.ts` — added unique timestamps to 4 Tim White profile mobile variants

## Verification

```
✅ 29 unit tests pass (pricing × 16, seed invariants × 9, timestamps × 4)
✅ TypeScript: no errors
✅ Biome: clean (1 pre-existing warning in OnboardingTurnstile.tsx, outside HOT ZONE)
✅ Marketing routes: revalidate = false (fully static)
✅ CodeRabbit: all 5 comments addressed (3 first-round + 2 second-round)
```

## Bot Reviews

| Bot | Status |
|-----|--------|
| CodeRabbit | ✅ All 5 comments resolved |
| Greptile | ✅ No open comments |

## Auto-merge

Eligible: this is a marketing/data/test-only change, not auth/billing/stripe.

## Summary by CodeRabbit

* **New Features**
  * Simplified pricing structure with Free, Pro, and Max tiers
  * Refreshed call-to-action messaging throughout the pricing page

* **Refactor**
  * Consolidated plan configuration for consistency across all pages
  * Enhanced pricing page schema for improved search engine visibility

[Review Change Stack](https://app.coderabbit.ai/change-stack/JovieInc/Jovie/pull/8451)

### CODE REVIEW

**Comment by itstimwhite** [comment]
## Review Summary
**Verdict**: Approve

This PR cleanly establishes a pricing source-of-truth chain, removes legacy plan tiers and waitlist copy, adds screenshot timestamp invariants, and enforces fixture data quality. The changes are well-structured, follow Jovie conventions, and include thorough contract tests.

---

### Critical Issues
None found.

---

### Warnings
- **`apps/web/data/marketingPricingPlans.ts:90-92`** — The function `getMarketingPlanHref` is still exported but no longer used anywhere in the codebase (the component now reads `plan.ctaHref` directly). This is dead code that could confuse future maintainers. Consider removing it.

- **`apps/web/constants/plans.ts:56,65,74`** — The `features` arrays are cast as `readonly string[]` from `ENTITLEMENT_REGISTRY.*.marketing.features`. If the entitlement registry’s feature arrays ever change their element type (e.g., to objects), this cast will silently pass incorrect data. Consider adding a runtime type guard or a stricter type assertion that validates the shape at test time.

---

### Suggestions
- **`apps/web/tests/unit/product-screenshots/screenshot-player-timestamps.test.ts:90-104`** — The test “keeps Tim White profile mobile variants with player as the canonical player timestamp source” hardcodes scenario IDs (`tim-white-profile-{listen,pay,live,subscribe}-mobile`). If these scenarios are renamed or removed, the test will fail with a cryptic “not found” error. Consider iterating over `SCREENSHOT_SCENARIOS` and checking that any scenario whose `id` matches a known pattern (e.g., `tim-white-profile-*-mobile`) and whose `queryKey` implies a visible player has a `playerTimestamp` defined. This would be more resilient to refactoring.

- **`apps/web/tests/unit/pricing/pricing-source-of-truth.test.ts`** — The test suite validates that `CANONICAL_PLANS` display names match the entitlement registry, but it does not validate that the **feature lists** match. Since `marketingPricingPlans.ts` now defines its own feature lists (e.g., “Up to 100 contacts” for free, “Everything in Free” for pro), a future change to the entitlement registry’s features could cause drift. Consider adding a contract test that asserts `MARKETING_PRICING_PLANS` features are a superset of (or identical to) the canonical plan features, or at least that no feature in the marketing plans contradicts the registry.

- **`apps/web/data/marketingPricingPlans.ts`** — The `MARKETING_PRICING_PLANS` array is declared `as const`, but the `features` arrays inside each plan are not individually `as const`. This means the inferred type of `features` is `string[]` rather than `readonly string[]`. While not a bug, making them `as const` would improve type safety and align with the `readonly` expectation in `MarketingPricingPlan`.

---

### What Looks Good
- **Clean separation of concerns**: The pricing chain (plan-prices → entitlement registry → `constants/plans.ts` → `marketingPricingPlans.ts` → UI) is well-documented and enforced by contract tests.
- **Removal of waitlist/team/enterprise**: The PR decisively removes legacy concepts, simplifying the pricing surface and eliminating stale copy.
- **Screenshot timestamp invariants**: The `playerTimestamp` field and its uniqueness test prevent copy-paste-looking marketing screenshots — a thoughtful quality gate.
- **Fixture invariant tests**: Catching placeholder titles and incorrect collaborator credits in seed data will prevent embarrassing demo content from reaching production.
- **All CTAs now include `?plan=<id>`**: This ensures onboarding can read the user’s intent, improving the signup flow.
- **No convention violations**: The PR respects Jovie’s rules (no emoji, no native dialogs, static marketing pages, TypeScript strictness, etc.).

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot (deepseek/deepseek-v4-flash).

File: apps/web/constants/plans.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+export const CANONICAL_PLANS: readonly CanonicalPlan[] = [
+  {
+    id: 'free',
+    displayName: ENTITLEMENT_REGISTRY.free.marketing.displayName,
+    tagline: ENTITLEMENT_REGISTRY.free.marketing.tagline,
+    monthlyPriceUsd: 0,
+    yearlyPriceUsd: null,
+    monthlyPriceLabel: '$0',
+    features: ENTITLEMENT_REGISTRY.free.marketing.features as readonly string[],
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/constants/plans.ts]: Fixed: removed the unnecessary `as readonly string[]` casts from all three plans in `CANONICAL_PLANS`. `ENTITLEMENT_REGISTRY.*.marketing.features` is already typed as `readonly string[]` so the assertion was redundant. Applied to free (line 56), pro (line 67), and max (line 78).

File: apps/web/tests/unit/product-screenshots/screenshot-player-timestamps.test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    const scenariosWithTimestamp = SCREENSHOT_SCENARIOS.filter(
+      scenario => scenario.playerTimestamp !== undefined
+    );
+
+    for (const scenario of scenariosWithTimestamp) {
+      expect(
+        TIMESTAMP_PATTERN.test(scenario.playerTimestamp as string),
+        `Scenario "${scenario.id}" has invalid playerTimestamp format: "${scenario.playerTimestamp}". Expected M:SS or MM:SS.`
+      ).toBe(true);
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/tests/unit/product-screenshots/screenshot-player-timestamps.test.ts]: Fixed: tightened the regex from `/^\d{1,2}:\d{2}$/` to `/^\d{1,2}:[0-5]\d$/` so the invariant rejects impossible seconds like 1:99. All 4 existing timestamp tests still pass.

File: apps/web/tests/unit/seed/seed-fixture-invariants.test.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  });
+
+  it('contains no known placeholder titles in release data', () => {
+    const fixtureText = fixtureJson({
+      releases: INTERNAL_DJ_DEMO_PERSONA.releases,
+    });
+
+    for (const placeholder of KNOWN_PLACEHOLDER_TITLES) {
+      expect(fixtureText).not.toContain(placeholder);
+    }
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/tests/unit/seed/seed-fixture-invariants.test.ts:65]: Fixed: normalized both the fixture JSON output and each `KNOWN_PLACEHOLDER_TITLES` entry to lowercase before comparison, making the check fully case-insensitive. All 9 fixture invariant tests still pass.

File: apps/web/app/(marketing)/launch/pricing/page.tsx [hunk 4]
```diff
@@ -92,10 +92,10 @@ export default function PricingPage() {
               <FeatureList features={free.marketing.features} />
             </div>
             <Link
-              href={APP_ROUTES.SIGNUP}
+              href={`${APP_ROUTES.SIGNUP}?plan=free`}
               className='focus-ring inline-block mt-7 px-5 py-2.5 rounded-md font-medium text-sm transition-colors border border-subtle hover:bg-white/[0.04]'
             >
-              Request Access
+              Claim your profile
             </Link>
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/app/(marketing)/launch/pricing/page.tsx]: Fixed —  now uses `getCanonicalPlan()` from `constants/plans.ts` for all three plan CTAs (label + href). Committed in 09c57c6.

File: apps/web/tests/unit/pricing/pricing-source-of-truth.test.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    const maxPlan = MARKETING_PRICING_PLANS.find(p => p.id === 'max');
+    expect(maxPlan?.price).toBe(`$${PLAN_PRICES.max.monthly}`);
+  });
+
+  it('does not include the legacy team or enterprise plan IDs', () => {
+    const planIds = MARKETING_PRICING_PLANS.map(p => p.id);
+    expect(planIds).not.toContain('team');
+    expect(planIds).not.toContain('enterprise');
+  });
+});
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/tests/unit/pricing/pricing-source-of-truth.test.ts:178]: Added — new `has non-empty features list for every plan` test in the same suite. MARKETING_PRICING_PLANS intentionally uses curated short-form marketing copy while CANONICAL_PLANS derives from the entitlement registry (different level of detail by design), so the test verifies both lists are non-empty rather than exact equality. Committed in 09c57c6.

**Comment by itstimwhite** [comment]
## Review Summary

**Verdict**: Approve

This is a well-structured PR that establishes a proper pricing source of truth, removes deprecated waitlist/request-access copy, and adds comprehensive contract tests. The changes are focused, well-documented, and follow Jovie conventions.

### Critical Issues

None found.

### Warnings

- **apps/web/data/marketingPricingPlans.ts:94-96** — The `getMarketingPlanHref` function is now unused since `ctaHref` is used directly. Consider removing it to prevent dead code accumulation. However, this is non-blocking as it doesn't affect functionality.

- **apps/web/constants/plans.ts:1-91** — The file header comment mentions "Never mention 'waitlist' in any plan label, CTA, or description" but this rule is enforced in tests rather than at the type level. Consider adding a TypeScript branded type or a lint rule to make this a compile-time guarantee rather than a runtime test. Non-blocking.

### Suggestions

- **apps/web/constants/plans.ts:85-87** — The `getCanonicalPlan` function accepts `string` instead of `CanonicalPlanId`. This allows passing invalid plan IDs at compile time. Consider narrowing the parameter type:

```typescript
export function getCanonicalPlan(id: CanonicalPlanId): CanonicalPlan | undefined {
  return CANONICAL_PLANS.find(plan => plan.id === id);
}
```

- **apps/web/tests/unit/pricing/pricing-source-of-truth.test.ts:96-103** — The test "monthly price label matches monthlyPriceUsd" only checks when `monthlyPriceUsd !== null`. Consider also testing that `monthlyPriceLabel` is `'$0'` for the free plan (where `monthlyPriceUsd` is `0`, not `null`).

- **apps/web/app/(marketing)/launch/pricing/page.tsx:43-45** — The fallback to `CANONICAL_PLANS[0/1/2]` is fragile if the array order changes. Consider using `getCanonicalPlan` with a non-null assertion or a `findOrThrow` helper:

```typescript
const freePlan = getCanonicalPlan('free')!;
const proPlan = getCanonicalPlan('pro')!;
const maxPlan = getCanonicalPlan('max')!;
```

### What Looks Good

1. **Single source of truth** — `constants/plans.ts` properly centralizes plan data, deriving from the entitlement registry and plan-prices. This prevents the copy/paste drift that plagues most pricing implementations.

2. **Comprehensive contract tests** — 16 tests covering plan IDs, prices, banned phrases, signup hrefs, and feature list integrity. The test for legacy plan IDs (`team`/`enterprise`) being absent is a nice migration guard.

3. **Clean removal of waitlist logic** — The `isMarketingPlanActive`/`getMarketingPlanCtaLabel`/`getMarketingPlanHref` helpers are properly removed, and the `NEXT_PUBLIC_MARKETING_ACTIVE_PLANS` env var is eliminated. This simplifies the codebase significantly.

4. **Screenshot timestamp invariants** — The `playerTimestamp` field and its uniqueness test prevent marketing screenshots from looking copy-pasted. Good attention to visual authenticity.

5. **Proper type safety** — `CanonicalPlan` interface uses `readonly` throughout, `as const` assertions on arrays, and the `CanonicalPlanId` union type.

6. **Clear import rules** — The file header in `constants/plans.ts` documents exactly which modules should import from where (server components vs marketing UI vs Stripe billing).

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot (deepseek/deepseek-v4-flash).

**Comment by itstimwhite** [comment]
## Review Summary

**Verdict**: Changes Requested

The PR introduces a well-structured pricing source of truth, removes outdated waitlist/request-access copy, and adds valuable contract tests. However, there is a critical versioning issue that must be resolved before merging.

---

### Critical Issues

- **`VERSION` line 1** — Version downgrade from `26.4.241` to `26.4.240`.  
  The diff shows the version file changing from `26.4.241` to `26.4.240`, which is a regression. The PR description states version `26.4.236`, but the actual change is to `26.4.240`. This is likely a merge conflict resolution error.  
  **Fix**: Revert the version to `26.4.241` (or increment to `26.4.242` if this PR represents a new release). Ensure the CHANGELOG entry matches the final version.

---

### Warnings

- **`apps/web/data/marketingPricingPlans.ts`** — The `getMarketingPlanHref` function has been removed.  
  The PR only updates `MarketingPricingPlans.tsx` to use `plan.ctaHref`, but if any other file imports `getMarketingPlanHref`, it will break. Please confirm no other references exist in the codebase (e.g., via a quick `grep`). If there are, update them or re-export the function.

- **`apps/web/data/marketingPricingPlans.ts`** — The `MarketingPricingPlan` interface still includes `accent: 'pink'` in its union type, but no plan currently uses it (the `max` plan uses `'violet'`).  
  While not a bug, this dead type member can cause confusion. Consider removing `'pink'` if it's no longer needed.

- **`apps/web/tests/unit/pricing/pricing-source-of-truth.test.ts`** — Hardcoded valid plan IDs (`['free', '

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot (deepseek/deepseek-v4-flash).
